### ISSUE TITLE
export 'useNavigate' (imported as 'Ki') was not found in 'react-router'

### ISSUE THREAD

**Issue by niksamoil** [marmelab/react-admin#7374]
**What happened:**
Compiled with problems:

ERROR in ./node_modules/ra-core/dist/esm/index.js 1200:10-12

export 'useNavigate' (imported as 'Ki') was not found in 'react-router' (possible exports: MemoryRouter, Prompt, Redirect, Route, Router, StaticRouter, Switch, generatePath, matchPath, withRouter)

* React-admin version: 4.0.0-beta.2
* Last version that did not exhibit the issue (if applicable):
* React version: 17.0.2
* Browser: Chrome Latest version
* Stack trace (in case of a JS error):

**Comment by WiXSL** [comment]
I can reproduce this kind of error.
Could you explain the steps to reproduce this?

**Comment by niksamoil** [comment]
@WiXSL 
I try to migrate my old RA from v2.9 to 4.0. 
I updated all required dependencies. Now I have only login component, dataProvider, authProvider, i18nProvider setup active, and error appears even from start of launching app.
I also opened this file and found that only this import is from react-router, but other like `useNavigate as Wi` or `useNavigate as xi` are imported from react-router-dom.

[screenshot](https://prnt.sc/TABO6to1dTM9)
